File: partition-go/partition-go/solution_v1.go
package partition_go

type Node struct {
	Next *Node
	Data int
}

func PartitionV1(node *Node, data int) *Node {
	var partA *Node
	var partB *Node
	for node != nil {
		if node.Data < data {
			partA = addInTail(partA, node)
		} else {
			partB = addInTail(partB, node)
		}
		node = node.Next
	}

	merged := merge(partA, partB)
	return merged
}

func merge(partA *Node, partB *Node) *Node {
	head := partA
	for {
		if partA.Next == nil {
			partA.Next = partB
			break
		}
		partA = partA.Next
	}

	return head
}

func addInTail(node *Node, current *Node) *Node {
	if node == nil {
		node = &Node{
			Next: nil,
			Data: current.Data,
		}
		return node
	}

	head := node
	for {
		if node.Next == nil {
			node.Next = &Node{
				Next: nil,
				Data: current.Data,
			}
			break
		}
		node = node.Next
	}

	return head
}

File: sum-list-java/sum-list-java/src/main/java/SolutionV1.java
public class SolutionV1 {

    Node sumList(Node nodeOne, Node nodeTwo) {
        StringBuilder sbOne = new StringBuilder();
        helper(nodeOne, sbOne);

        StringBuilder sbTwo = new StringBuilder();
        helper(nodeTwo, sbTwo);
        String result = String.valueOf(Integer.parseInt(sbOne.toString()) + Integer.parseInt(sbTwo.toString()));

        Node previous;
        if (result.length() == 0) {
            return null;
        }
        previous = new Node(null, Character.getNumericValue(result.charAt(result.length() - 1)));
        Node head = previous;

        for (int i = result.length() - 2; i >= 0; i--) {
            Node current = new Node(null, Character.getNumericValue(result.charAt(i)));
            previous.next = current;
            previous = current;
        }
        return head;
    }

    void helper(Node node, StringBuilder sb) {
        if (node.next == null) {
            sb.append(node.data);
            return;
        }

        helper(node.next, sb);
        sb.append(node.data);
    }
}
